plugins/rsyslog.yaml

```diff
@@ -1,21 +1,41 @@
 # Plugin Info
-version: 0.0.3
+version: 0.0.4
 title: Rsyslog
 description: Log parser for Rsyslog
 parameters:
+  - name: source
+    label: Log source
+    description: Use this field to specify where your logs are coming from. When choosing the 'file' option, the agent reads in logs from the log paths specified below. When choosing the 'tcp' or 'udp' options, the agent will listen on specified port.
+    type: enum
+    valid_values:
+      - file
+      - udp
+      - tcp
+    default: udp
   - name: listen_port
     label: Listen Port
     description: A port which the agent will listen for syslog messages
     type: int
     default: 514
-  - name: connection_type
-    label: Connection Type
-    description: The type of Rsyslog connection (`udp` or `tcp`)
```

Comment:
Removing this breaks backwards compatibility, I think we should
- make this param hidden
- if set, overwrite `source` if `source` is not set

Does that make sense? Maybe easier said than done.

Reply:
This is a bit of a challenge, but in my testing this seems to work.
```
# {{ if or (eq $source "udp") (eq $connection_type "udp") }}
# {{ if ne $connection_type "tcp" }}
# {{ if ne $source "file" }}
  - id: rsyslog_udp_input
    type: udp_input
    listen_address: '{{ if eq $length 0 }}{{ $listen_ip }}:{{ $listen_port }}{{ else }}{{ $listen_address }}{{ end }}'
    labels:
      log_type: rsyslog
      plugin_id: {{ .id }}
    add_labels: true
    output: rsyslog_parser
# {{ end }}
# {{ end }}
# {{ end }}
```